### PR TITLE
Use SINK INTO for GenericMapStore store operations [HZ-1639]

### PR DESCRIPTION
**Pull request by orcunc**
The original PR is split into two. 
The first PR  - https://github.com/hazelcast/hazelcast/pull/23416 - is related to SQL layer
The second PR - this one - is related to GenericMapStore 

The GenericMapstore is using "SINK INTO" Hazelcast SQL instead of "INSERT INTO". The SINK INTO clause is parsed by Calcite and the call triggers JdbcSqlConnector#sinkProcessor

Jira : https://hazelcast.atlassian.net/browse/HZ-1639

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

**Review by frant-hartm — APPROVED** [review]

File: extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapStore.java [hunk 1]
```diff
@@ -583,9 +550,7 @@ private static class GenericMapStoreProperties {
 
             String columnsProperty = properties.getProperty(COLUMNS_PROPERTY);
             if (columnsProperty != null) {
-                List<String> columns = new ArrayList<>();
-                Collections.addAll(columns, columnsProperty.split(","));
-                this.columns = unmodifiableList(columns);
+                this.columns = Arrays.asList(columnsProperty.split(","));
```
[the remark this answers is not in the document]
Comment (orcunc) [reply extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapStore.java:553]: Ok

File: extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapStore.java [hunk 2]
```diff
@@ -560,6 +497,36 @@ void awaitInitFinished() {
...
+
+    // SQLException returns SQL state in five digit number.
+    // These five-digit numbers tell about the status of the SQL statements.
+    // The SQLSTATE values consists of two fields.
+    // The class, which is the first two characters of the string, and
+    // the subclass, which is the terminating three characters of the string.
+    // See https://en.wikipedia.org/wiki/SQLSTATE for cate
+    boolean isIntegrityConstraintViolation(Exception exception) {
+        boolean result = false;
+        Throwable rootSQLException = findSQLException(exception);
```
[the remark this answers is not in the document]
Comment (orcunc) [reply extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapStore.java:519]: Ok

File: extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreMSSQLIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+
+        ClientConfig clientConfig = new ClientConfig();
+
+        initializeWithClient(2, config, clientConfig);
+        sqlService = instance().getSql();
+    }
+
+
+
```
[the remark this answers is not in the document]
Comment (orcunc) [reply extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreMSSQLIntegrationTest.java:62]: Ok

File: extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreMSSQLIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import com.hazelcast.test.jdbc.MSSQLDatabaseProvider;
+import org.junit.BeforeClass;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static org.assertj.core.util.Lists.newArrayList;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class})
+public class GenericMapStoreMSSQLIntegrationTest extends GenericMapStoreIntegrationTest {
```
[the remark this answers is not in the document]
Comment (orcunc) [reply extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreMSSQLIntegrationTest.java:36]: Ok

File: extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreMSSQLIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.mapstore;
```
[the remark this answers is not in the document]
Comment (orcunc) [reply extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreMSSQLIntegrationTest.java:17]: Moved to mssql package

File: extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreMSSQLIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+                        .setClassName(JdbcDataStoreFactory.class.getName())
+                        .setProperty("jdbcUrl", dbConnectionUrl)
+        );
+
+
+        ClientConfig clientConfig = new ClientConfig();
+
+        initializeWithClient(2, config, clientConfig);
+        sqlService = instance().getSql();
+    }
```
[the remark this answers is not in the document]
Comment (orcunc) [reply extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreMSSQLIntegrationTest.java:59]: Moved the code into base test class

File: extensions/mapstore/src/main/java/com/hazelcast/mapstore/Queries.java [hunk 7]
```diff
@@ -43,36 +43,34 @@ class Queries {
...
         String columnNames = columnMetadata.stream()
-                                           .map(sqlColumnMetadata -> '\"' + sqlColumnMetadata.getName() + '\"')
-                                           .collect(joining(", "));
+                .map(sqlColumnMetadata -> '\"' + sqlColumnMetadata.getName() + '\"')
+                .collect(joining(", "));
 
-        storeInsert = "INSERT INTO \"" + mapping + "\" ( " + columnNames + " ) VALUES (" +
-                queryParams(columnMetadata.size()) +
-                ")";
+        storeInsert = String.format("SINK INTO \"%s\" (%s) VALUES (%s)", mapping, columnNames,
```
[the remark this answers is not in the document]
Comment (orcunc) [reply extensions/mapstore/src/main/java/com/hazelcast/mapstore/Queries.java:57]: Renamed as storeSink

File: extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreIntegrationTest.java [hunk 8]
```diff
@@ -207,29 +205,6 @@ public void testMapClear() {
...
-        })
-                .isInstanceOf(HazelcastSerializationException.class)
-                .hasMessageContaining("Invalid field kind: 'id for Schema" +
-                        " { className = org.example.Person, numberOfComplexFields = 2," +
-                        " primitivesLength = 0, map = {name=FieldDescriptor{" +
-                        "name='name', kind=STRING, index=1, offset=-1, bitOffset=-1}," +
-                        " id=FieldDescriptor{name='id', kind=STRING, index=0, " +
-                        "offset=-1, bitOffset=-1}}}, valid field kinds : " +
-                        "[INT32, NULLABLE_INT32], found : STRING");
-    }
```
[the remark this answers is not in the document]
Comment (orcunc) [reply extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreIntegrationTest.java]: Changed the test to show that the method parameter is used and the string in GenericRecord is ignored.

File: extensions/mapstore/src/test/java/com/hazelcast/mapstore/QueriesTest.java [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+        Queries queries = new Queries(mapping, idColumn, columnMetadata);
+        String result = queries.load();
+        Assert.assertEquals("SELECT * FROM \"mymapping\" WHERE \"id\" = ?", result);
+    }
+
+    @Test
+    public void testLoadAll() {
+        Queries queries = new Queries(mapping, idColumn, columnMetadata);
+        String result = queries.loadAll(2);
+        Assert.assertEquals("SELECT * FROM \"mymapping\" WHERE \"id\" IN (?, ?)", result);
```
Comment (TomaszGaweda) [inline extensions/mapstore/src/test/java/com/hazelcast/mapstore/QueriesTest.java:50]: nit: for clarity you can add static import for assertEquals

**Review by TomaszGaweda — APPROVED** [review]

File: extensions/mapstore/src/test/java/com/hazelcast/mapstore/QueriesTest.java (hunk 9 above)
Comment (orcunc) [reply extensions/mapstore/src/test/java/com/hazelcast/mapstore/QueriesTest.java:50]: Ok

**Pull request merged**